Add R2 bucket lock changelog entry + update bucket lock docs (#20566)
cloudflare/cloudflare-docs: src/content/changelog/r2/2025-03-06-r2-bucket-locks.mdx

---
title: Set retention polices for your R2 bucket with bucket locks
description: Set retention polices for your R2 bucket with bucket locks
products:
  - r2
date: 2025-03-06T10:00:00Z
---

You can now use [bucket locks](/r2/buckets/bucket-locks/) to set retention policies on your [R2 buckets](/r2/buckets/) (or specific prefixes within your buckets) for a specified period — or indefinitely. This can help ensure compliance by protecting important data from accidental or malicious deletion.

> 

Reviewer: ```suggestion

Locks allow you to:

- Lock objects for a specific duration. For example, 90 days.
- Retain objects until a certain date. For example, until January 1, 2030.
- Keep objects locked indefinitely.
```

Reviewer: contextualize what a lock is sooner

> Buckets can have up to 1,000 [bucket lock rules](/r2/buckets/). Each rule specifies which objects it covers (via prefix) and how long those objects must remain retained. You can:

Reviewer: ```suggestion
Buckets can have up to 1,000 [bucket lock rules](/r2/buckets/). Each rule specifies which objects it covers (via prefix) and how long those objects must remain retained.
```

> - Lock objects for a specific duration. For example, 90 days.
> - Retain objects until a certain date. For example, until January 1, 2030.
> - Keep objects locked indefinitely.

Reviewer: ```suggestion
```

Here's how you can add a bucket lock rule with [Wrangler](/workers/wrangler/):

```sh

> npx wrangler r2 bucket lock add my-bucket

Reviewer: ```suggestion
# Prevent deletion of objects in the bucket before 180 days
npx wrangler r2 bucket lock add <bucket> 180-days --retention-days=180

# Configure a rule for a subset of objects matching a prefix
npx wrangler r2 bucket lock add <bucket> indefinite-logs "logs" --retention-indefinite
```

Can we show a prefix or example rule file here?

Also - the API is a little weird. What is an ID? What is a valid ID? Why not generate it for the user?

```
‚ûú  npx wrangler r2 bucket lock add --help
wrangler r2 bucket lock add <bucket> [id] [prefix]

Add a lock rule to an R2 bucket

POSITIONALS
  bucket  The name of the R2 bucket to add a bucket lock rule to  [string] [required]
  id      A unique identifier for the bucket lock rule  [string]
  prefix  Prefix condition for the bucket lock rule (set to "" for all prefixes)  [string]
```

---
title: Set retention polices for your R2 bucket with bucket locks
description: Set retention polices for your R2 bucket with bucket locks
products:
  - r2
date: 2025-03-06T10:00:00Z
---

You can now use [bucket locks](/r2/buckets/bucket-locks/) to set retention policies on your [R2 buckets](/r2/buckets/) (or specific prefixes within your buckets) for a specified period — or indefinitely. This can help ensure compliance by protecting important data from accidental or malicious deletion.

Locks allow you to:

> - Lock objects for a specific duration. For example, 90 days.

Reviewer 2: consistency between lock object vs. retain object?

> - Lock objects for a specific duration. For example, 90 days.

Reviewer: ```suggestion
- Lock objects for a specific duration: for example, preventing object deletion until 90 days.
```

> - Lock objects for a specific duration. For example, 90 days.

Reviewer 3: ```suggestion
- Lock objects against deletion for a specific duration, for example, for 90 days.
```

> - Retain objects until a certain date. For example, until January 1, 2030.

Reviewer 3: ```suggestion
- Lock objects against deletion until a specific date. For example, until January 1, 2030.
```

> - Keep objects locked indefinitely.

Reviewer 3: ```suggestion
- Lock objects against deletion indefinitely.
```
